integration_tests/test_math.py

```diff
@@ -72,13 +72,18 @@ def test_ldexp():
 
 
 def test_fabs():
-    i: f64
-    j: f64
+
     eps: f64
     eps = 1e-12
-    i = fabs(10.3)
-    j = fabs(-10.3)
-    assert abs(i - j) < eps
+    i: f64
+    j: f64
+    i = -10.5
+    j = -7.0
+    assert fabs(i + fabs(j)) == 3.5
+    assert abs(fabs(10) - fabs(-10)) < eps
+    assert abs(fabs(10.3) - fabs(-10.3)) < eps
+    assert fabs(0.5) == 0.5
+    assert fabs(-5) == 5.0
```

Comment:
It seems to me that you are not testing these functions properly. `fabs(x: i64)` according to your implementation below returns an integer. You should try the runtime values (non-constant values):

```diff
diff --git a/integration_tests/test_math.py b/integration_tests/test_math.py
index 6e17c5bd1..970c569cb 100644
--- a/integration_tests/test_math.py
+++ b/integration_tests/test_math.py
@@ -83,7 +83,9 @@ def test_fabs():
     assert abs(fabs(10) - fabs(-10)) < eps
     assert abs(fabs(10.3) - fabs(-10.3)) < eps
     assert fabs(0.5) == 0.5
-    assert fabs(-5) == 5.0
+    _i: i64
+    _i = 5
+    assert fabs(i) == 5.0
```
This test will not pass.